Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,317 @@
+        }
+
+        return headers;
+    }
+
+    /// <summary>
+    /// Generates a random X-Amzn-Trace-Id for REST API mode.
+    /// </summary>
+    /// <returns>A string representing a random X-Amzn-Trace-Id in the format used by API Gateway for REST APIs.</returns>
+    /// <remarks>
+    /// The generated trace ID includes:
+    /// - A root ID with a timestamp and two partial GUIDs
+    /// - A parent ID
+    /// - A sampling decision (always set to 0 in this implementation)
+    /// - A lineage identifier
+    /// </remarks>
+    private static string GenerateTraceId()
+    {
+        var timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds().ToString("x");
+        var guid1 = Guid.NewGuid().ToString("N");
+        var guid2 = Guid.NewGuid().ToString("N");
+        return $"Root=1-{timestamp}-{guid1.Substring(0, 12)}{guid2.Substring(0, 12)};Parent={Guid.NewGuid().ToString("N").Substring(0, 16)};Sampled=0;Lineage=1:{Guid.NewGuid().ToString("N").Substring(0, 8)}:0";
+    }
+
+
+    /// <summary>
+    /// Generates a random API Gateway request ID for HTTP API v1 and v2.
+    /// </summary>
+    /// <returns>A string representing a random request ID in the format used by API Gateway for HTTP APIs.</returns>
+    /// <remarks>
+    /// The generated ID is a 145character string consisting of lowercase letters and numbers, followed by an equals sign.
+    private static string GenerateRequestId()
+    {
+        return Guid.NewGuid().ToString("N").Substring(0, 8) + Guid.NewGuid().ToString("N").Substring(0, 7) + "=";
+    }
+
+    /// <summary>
+    /// Sets the response body on the HttpResponse.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the body on.</param>
+    /// <param name="body">The body content.</param>
+    /// <param name="isBase64Encoded">Whether the body is Base64 encoded.</param>
+    private static void SetResponseBody(HttpResponse response, string? body, bool isBase64Encoded, ApiGatewayEmulatorMode apiGatewayEmulator)
+    {
+        if (!string.IsNullOrEmpty(body))
+        {
+            byte[] bodyBytes;
+            if (isBase64Encoded && ApiGatewayEmulatorMode.Rest != apiGatewayEmulator) // rest api gateway doesnt automatically decode the response
+            {
+                bodyBytes = Convert.FromBase64String(body);
+            }
+            else
+            {
+                bodyBytes = Encoding.UTF8.GetBytes(body);
+            }
+
+            response.Body = new MemoryStream(bodyBytes);
+            response.ContentLength = bodyBytes.Length;
+        }
+    }
+
+    /// <summary>
+    /// Sets the content type and status code for API Gateway v1 responses.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the content type and status code on.</param>
+    /// <param name="headers">The single-value headers.</param>
+    /// <param name="multiValueHeaders">The multi-value headers.</param>
+    /// <param name="statusCode">The status code to set.</param>
```

Comment:
ApiGatewayEmulatorMode emulatorMode does not have a docs param

Reply:
fixed